js_modules/dagster-ui/packages/ui-core/src/app/InjectedComponentContext.tsx

```diff
@@ -18,30 +23,57 @@ type AComponentWithProps<Props = Record<string, never>> =
 
 type InjectedComponentContextType = {
   AppTopNavRightOfLogo: AComponentFromComponent<typeof AppTopNavRightOfLogo> | null;
-  OverviewPageAlerts?: AComponentWithProps | null;
+  OverviewPageAlerts: AComponentWithProps | null;
   UserPreferences: AComponentFromComponent<typeof UserPreferences> | null;
   AssetsOverview: AComponentFromComponent<typeof AssetsOverviewRoot> | null;
   FallthroughRoot: AComponentFromComponent<typeof FallthroughRoot> | null;
-  AssetGraphHeader: AComponentFromComponent<typeof AssetGraphHeader> | null;
+  AssetsGraphHeader: AComponentFromComponent<typeof AssetsGraphHeader> | null;
 
-  RunMetricsDialog?: AComponentWithProps<{
+  RunMetricsDialog: AComponentWithProps<{
     runId: string;
     isOpen: boolean;
     onClose: () => void;
   }> | null;
+  AssetPageHeader: AComponentFromComponent<typeof AssetPageHeader>;
 };
-export const InjectedComponentContext = React.createContext<InjectedComponentContextType>(
-  {} as any,
-);
+
+type InjectedHookContextType = {
+  useAssetDefinitionFilterState: typeof useAssetDefinitionFilterState;
+  useAssetCatalogFiltering: typeof useAssetCatalogFiltering;
+  useAssetGraphExplorerFilters: typeof useAssetGraphExplorerFilters;
+};
+export const InjectedComponentContext = React.createContext<{
+  components: InjectedComponentContextType;
+  hooks: InjectedHookContextType;
+}>({components: {} as any, hooks: {} as any});
 
 export function componentStub<TComponentKey extends keyof InjectedComponentContextType>(
-  component: TComponentKey,
+  componentName: TComponentKey,
 ): NonNullable<InjectedComponentContextType[TComponentKey]> {
   return (props: any) => {
-    const {[component]: Component} = useContext(InjectedComponentContext);
+    const {
+      components: {[componentName]: Component},
+    } = useContext(InjectedComponentContext);
     if (Component) {
       return <Component {...props} />;
     }
     return null;
   };
 }
+
+export function hookStub<TFunctionKey extends keyof InjectedHookContextType>(
+  hookName: TFunctionKey,
+): InjectedHookContextType[TFunctionKey] {
+  return function useHookStub(...args: Parameters<InjectedHookContextType[TFunctionKey]>) {
+    const {
+      hooks: {[hookName]: hook},
+    } = useContext(InjectedComponentContext);
+
+    if (!hook) {
+      assertUnreachable(hook);
+    }
+
+    // @ts-expect-error - "A spread argument must either have a tuple type or be passed to a rest parameter.ts", not sure why we get this error
+    return hook(...args);
+  } as InjectedHookContextType[TFunctionKey];
+}
```

Comment:
Only somewhat related, but while we're here, I don't think this is a "Stub"? I always think of a stub is an empty placeholder / mocked component. This is actually just a shorthand for using an injected component and doing the lookup on the fly. (nothing is "stubbed out")

I think something like `resolvedComponent('UserPreferences')` might explain what it's doing better